apps/examples/st_things/msghandling/sample_msghandling_user.c

```diff
@@ -0,0 +1,184 @@
+*
+* Copyright 2017 Samsung Electronics All Rights Reserved.
+*
+*
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*	  http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+******************************************************************/
+
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include <signal.h>
+#include <st_things/st_things.h>
+
+#define TAG	"MSG_HANDLING_USER"
+
+#define MAX_STR_LEN                                 64
+
+#define PROPERTY_VALUE_SWITCH 				        "power"
+
+#define PROP_DOORSTATE                              "doorState"
+#define PROP_AIRQUALITY                             "airQuality"
+#define PROP_CT                                     "ct"
+#define PROP_RANGE                                   "range"
+#define PROP_TONE                                   "toneValue"
+#define PROP_MACHINESTATES                          "machineStates"
+#define PROP_CURRENTMACHINESTATE                    "currentMachineState"
+
+static char *g_power_status[]                       = { "on", "off" };
+
+static bool g_switch_value                          = false;
+static char g_door_state[MAX_STR_LEN]               = "closed";
+static bool g_tone                                  = false;
+
+static double g_airqulity                           = 10.0;
+static double g_airqulity_range[MAX_STR_LEN]        = {0., 20.};
+
+static int64_t g_color_temp                         = 50;
+static int64_t g_color_temp_range[MAX_STR_LEN]      = {0, 100};
+
+static char g_machine_state[MAX_STR_LEN]            = "RUNNING";
+
+bool handle_get_request_on_switch(st_things_get_request_message_s *req_msg, st_things_representation_s *resp_rep)
+{
+	if (req_msg->has_property_key(req_msg, PROPERTY_VALUE_SWITCH)) {
+		resp_rep->set_str_value(resp_rep, PROPERTY_VALUE_SWITCH, g_switch_value ? g_power_status[0] : g_power_status[1]);
+	}
+	return true;
+}
+
+bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep)
+{
+
+	char *power;
+	if (req_msg->rep->get_str_value(req_msg->rep, PROPERTY_VALUE_SWITCH, &power)) {
+		if (strcmp(power, "off") == 0) {
+			g_switch_value = false;
+		} else {
+			g_switch_value = true;
+		}
+		resp_rep->set_str_value(resp_rep, PROPERTY_VALUE_SWITCH, g_switch_value ? g_power_status[0] : g_power_status[1]);
+		st_things_notify_observers(req_msg->resource_uri);
+	}
+
+	return true;
+}
+
+bool handle_get_request_on_resource_doorcontrol(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep)
```

Comment:
please check coding rule of * posistion.
st_things_get_request_message_s *req_msg